airflow/jobs.py

```diff
@@ -1434,7 +1434,7 @@ def _execute_helper(self, processor_manager):
                                                           State.NONE)
 
                 self._execute_task_instances(simple_dag_bag,
-                                             (State.SCHEDULED,))
```

Comment:
Sorry thinking about it, the better fix is to have it UP_FOR_RETRY. This way the scheduler and the task keep to their contract. It is not a race condition but a logic error.